### PR TITLE
Update account state.

### PR DESCRIPTION
**Pull request by dzeckelev**
Resolves BV-948.

### CODE REVIEW

File: doc/ui/rpc.md [hunk 1]
```diff
@@ -184,6 +184,36 @@ curl -X POST -H "Content-Type: application/json" --data '{"method": "ui_transfer
 ```
 </details>
 
+#### Update account status
+
+*Method*:	`updateAccountStatus`
+
+*Description*: Update account state.
```
Comment (furkhat) [inline doc/ui/rpc.md]: Suggest to leave the word "state". Just `*Description*: Updates an account.`

File: ui/account.go [hunk 2]
```diff
@@ -309,3 +309,30 @@ func (h *Handler) UpdateBalance(password, account string) error {
 
 	return nil
 }
+
+// UpdateAccountStatus updates account state.
+func (h *Handler) UpdateAccountStatus(password, account, name string,
```
Comment (furkhat) [inline ui/account.go]: Lets name it simply `UpdateAccount`, because it seems like it. Right?

File: ui/account_test.go [hunk 3]
```diff
@@ -353,3 +353,43 @@ func TestUpdateBalance(t *testing.T) {
...
+	}
+
+	for name, exp := range testNames {
+		res = handler.UpdateAccountStatus(
+			data.TestPassword, fxt.Account.ID, name, false, false)
+		assertMatchErr(nil, res)
+
+		db.Reload(fxt.Account)
+
+		if fxt.Account.Name != exp {
```
Comment (furkhat) [inline ui/account_test.go]: You could test all fields update within this loop

File: ui/account.go (hunk 2 above)
Comment (furkhat) [reply ui/account.go]: Accounts name cannot be described as status. There isn't "state" in the name. I still suggest to rename it as above.

File: doc/ui/rpc.md (hunk 1 above)
Comment (dzeckelev) [reply doc/ui/rpc.md]: I think that the word "state" is needed here. I cannot update any account field using this method.

File: ui/account_test.go [hunk 4]
```diff
@@ -353,3 +353,43 @@ func TestUpdateBalance(t *testing.T) {
...
+
+	res := handler.UpdateAccountStatus(
+		"wrong-password", fxt.Account.ID, "", false, false)
+	assertMatchErr(ui.ErrAccessDenied, res)
+
+	oldName := fxt.Account.Name
+	newName := util.NewUUID()[:30]
+
+	testNames := map[string]string{
+		"":      oldName,
```
Comment (ababo) [inline ui/account_test.go]: This looks suspicious.

File: doc/ui/rpc.md (hunk 1 above)
Comment (furkhat) [reply doc/ui/rpc.md]: It updates `name`, `in_use` and `is_default` - all of which are account fields. I don't understand what you mean.

File: ui/account_test.go [hunk 5]
```diff
@@ -353,3 +353,43 @@ func TestUpdateBalance(t *testing.T) {
...
+		}
+	}
+
+	res = handler.UpdateAccountStatus(
+		data.TestPassword, fxt.Account.ID, "", true, true)
+	assertMatchErr(nil, res)
+
+	db.Reload(fxt.Account)
+
+	if fxt.Account.InUse != true || fxt.Account.IsDefault != true {
```
Comment (ababo) [inline ui/account_test.go]: Why not `!fxt.Account.InUse || !fxt.Account.IsDefault`?

File: ui/account_test.go (hunk 4 above)
Comment (dzeckelev) [reply ui/account_test.go]: What confuses you? There is a check, if a name is empty, then it should not be changed.

**Comment by dzeckelev** [comment]
@furkhat @ababo updated.

**Pull request merged**